src/main/java/org/cactoos/experimental/Threads.java

```diff
@@ -23,13 +23,64 @@
  */
 package org.cactoos.experimental;
 
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import org.cactoos.Func;
+import org.cactoos.Scalar;
+import org.cactoos.collection.Mapped;
+import org.cactoos.iterable.IterableEnvelope;
+import org.cactoos.iterable.IterableOf;
+
 /**
  * Allows to execute the tasks concurrently.
  *
- * @param <X> The type of result objects.
- *
+ * @param <T> The type of task result item.
  * @since 1.0.0
+ * @todo #972:30min j.u.c.ExecutorService#invokeAll(java.util.Collection) should
+ *  be invoked with timeout. User should have the opportunity to pass the
+ *  timeout argument. We need a new decorator which supports the timeout.
+ * @todo #972:30min Add a new implementation which receive the tasks and able:
```

Comment:
I'm proposing a new object which will receive only tasks to the constructor.
Under the hood, the new object will create an executor service, execute the tasks and close them.
The user may have the ability to pass the timeout for the executions.

Reply:
@dgroup why can't that be another ctor of this `Threads` class?

Reply:
@llorllale, yes, we can create another `ctor` where will create a new ExecutorService (ES) instance, however, the ES should be shut down once the tasks are completed.

It means that we need to differentiate the cases when 
1. ES is passed outside of Threads and we sholdn't shutdown
2. ES is created within the `ctor` and should be shutdown

It may cause additional complexity, thus it's better to have separate classes, for example
1. Threads
    ```java
    public Threads(final ExecutorService exc, final Scalar<T>... tasks) {
        this(exc, new IterableOf<>(tasks));
    }
    ```
2. ClosableThreads, which may be based on `Threads`
    ```java
    public ClosableThreads(final Scalar<T>... tasks) {
        this(exc, new IterableOf<>(tasks));
    }
    ```

Reply:
@llorllale, I think I found the possible solution, how we can merge both these scenarios without huge complexity:
1. Move the current behavior (execute concurrently) to `ThreadsEnvelope` 
2. The `ThreadsEnvelope` should have the constructor which receive the tasks and the mapping function
3. The `Threads` will have several `ctor(s)` which have this functionality

Reply:
@llorllale done.